### PR TITLE
WIP: Run selftests on CI

### PR DESCRIPTION
**Pull request by ingenieroariel**
Do not merge - need to see CI output to compare with local.

### CODE REVIEW

File: lib/selftest/src/Obelisk/SelfTest.hs [hunk 1]
```diff
@@ -164,16 +164,16 @@ main = do
...
-          then it "can build ios" $ inTmpObInit $ \_ -> nixBuild ["-A", "ios.frontend"]
-          else it "can build android after accepting license" $ inTmpObInit $ \dir -> do
-            let defaultNixPath = dir </> ("default.nix" :: Shelly.FilePath)
-            writefile defaultNixPath
-              =<< T.replace
-                "# config.android_sdk.accept_license = false;"
-                "config.android_sdk.accept_license = true;"
-              <$> readfile defaultNixPath
-            nixBuild ["-A", "android.frontend"]
+        -- if System.Info.os == "darwin"
```
Comment (414owen) [inline lib/selftest/src/Obelisk/SelfTest.hs]: Just adding a reminder to reenable this before merging

File: release.nix [hunk 2]
```diff
@@ -3,12 +3,12 @@
     iosSdkVersion = "10.2";
   }
 , local-self ? import ./. self-args
+, cacheBuildSystems ? [ "x86_64-linux" "x86_64-darwin" ]
```
Comment (Ericson2314) [inline release.nix]: We should at some point rename these `supportedPlatforms` for consistency with Nixpkgs.

Comment (3noch) [reply release.nix]: `supportedSystems`

**Comment by ingenieroariel** [comment]
Closing this as it requires a lot of other work before it can pass and there is a different nixos style tests implementation.

**Comment by 3noch** [comment]
Great work on the NixOS style tests!

**Pull request closed without merging**